File: tests/characterization/fixtures/tier2/Pricing.swift
import Foundation

let taxRate = 0.2

func applyTax(amount: Double, rate: Double = taxRate) -> Double {
    // The tax is applied on the whole subtotal.
    return amount * (1 + rate)
}

protocol Priceable {
    func price() -> Double
}

class Invoice: Priceable {
    private let subtotal: Double

    init(subtotal: Double) {
        self.subtotal = subtotal
    }

    func price() -> Double {
        return applyTax(amount: subtotal, rate: taxRate)
    }
}

File: tests/characterization/fixtures/tier1/Pricing.java
package app.billing;

import app.money.Currency;

public interface Priceable {
    double price();
}

class Invoice implements Priceable {
    static final double TAX_RATE = 0.2;

    private final double subtotal;

    Invoice(double subtotal) {
        this.subtotal = subtotal;
    }

    public double price() {
        return applyTax(subtotal, TAX_RATE);
    }

    static double applyTax(double amount, double rate) {
        // The tax is applied on the whole subtotal.
        return amount * (1 + rate);
    }
}
